test/api.js

```diff
@@ -76,6 +76,15 @@ describe('api', function() {
       });
     });
 
+    it('should NOT compile empty data string', function(done) {
+      sass.render({
+        data: ''
+      }, function(error) {
+        assert.equal(error.message, 'Data context created with empty source string');
```

Comment:
IMO this error message is the real problem. We need to validate better in the CLI and return and error message that helps the user.